pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -270,9 +270,26 @@ func (s *Service) getOrCreateAPILoadBalancer(openStackCluster *infrav1.OpenStack
 		return nil, fmt.Errorf("network is not yet available in OpenStackCluster.Status")
 	}
 
-	// Create the VIP on the first cluster subnet
-	subnetID := openStackCluster.Status.Network.Subnets[0].ID
-	s.scope.Logger().Info("Creating load balancer in subnet", "subnetID", subnetID, "name", loadBalancerName)
+	var vipNetworkID, vipSubnetID string
+	if openStackCluster.Status.APIServerLoadBalancer != nil {
+		if openStackCluster.Status.APIServerLoadBalancer.LoadBalancerNetwork != nil {
```

Comment:
If we made the suggested change above where we copy in the cluster network when resolving the vip network, this block becomes unconditional. Also, when we add dual stack support here we get it in the non-custom case too, for free. You'll just want to return an error if APIServerLB or APIServerLB.Network are nil in the status, which will also mean we don't need all the nesting.

Reply:
good call, should be fixed now if I am not mistaken